apps/aechannel/src/aesc_fsm.erl

```diff
@@ -3071,14 +3128,56 @@ close_(Reason, D) ->
     end,
     {stop, Reason, D}.
 
-handle_call(St, Req, From, #data{} = D) ->
+handle_call(_, {?CONNECT_CLIENT, Pid, Tx} = Msg, From,
+            #data{ client_connected = false
```

Comment:
If someone tries to reconnect while the client is connected then this will crash the FSM, someone can use this to DOS the FSM.
Let's say Alice and Mallory uses the same node for state channels. Mallory want's to DOS Alice - all she needs to do is just constantly trying to reconnect to Alice's FSM - this will crash the FSM every time Alice reestablishes the channel.

Reply:
It pattern-matches on `client = undefined`, so this clause should not execute if a client is connected. But there should be some test code verifying this.

Reply:
Code has been added to the `aesc_fsm_SUITE` to check that trying to reconnect while a client is already connected yields an error. Currently, the error will be `unknown_request`.